src/Xhgui/Profile.php

```diff
@@ -398,6 +398,49 @@ public function sort($dimension, $data)
         uasort($data, $sorter);
         return $data;
     }
+    
+    /**
+     * @param array $profileData
+     * @param array $filters
+     *
+     * @return array
+     */
+    public function filter($profileData, $filters = [])
+    {
+        $filter = function ($item, $key) use ($filters) {
+            $nameToFilter = $this->getNameToFilter($key);
+            return !in_array($nameToFilter, $filters);
+        };
```

Comment:
If you `fnmatch` the profile data below, why do you also need to pre-filter here?

Reply:
I was thinking the same, today I mixed the new feature with the previous one but yes can be optimized.